File: Actito.UserInbox.Core/Models/ActitoUserInboxResponse.cs
namespace ActitoSdk.UserInbox.Core.Models;

/// <summary>
/// Represents the response returned when fetching a user's inbox.
/// </summary>
/// <remarks>
/// An <see cref="ActitoUserInboxResponse"/> contains the total number of inbox items,
/// the number of unread items, and the list of items themselves.
/// </remarks>
public class ActitoUserInboxResponse
{
    /// <summary>
    /// Total number of items in the user's inbox.
    /// </summary>
    public int Count { get; }

    /// <summary>
    /// Number of unread items in the user's inbox.
    /// </summary>
    public int Unread { get; }

    /// <summary>
    /// List of inbox items for the user.
    /// </summary>
    public IList<ActitoUserInboxItem> Items { get; }

    /// <summary>
    /// Constructor for <see cref="ActitoUserInboxResponse"/>.
    /// </summary>
    public ActitoUserInboxResponse(int count, int unread, IList<ActitoUserInboxItem> items)
    {
        Count = count;
        Unread = unread;
        Items = items;
    }
}

File: Actito.UserInbox.iOS.Binding.Bridge/Sources/Models/ActitoUserInboxResponse.swift
import Foundation
import ActitoUserInboxKit

@objc
public class ActitoUserInboxResponse : NSObject {
    @objc public let count: Int
    @objc public let unread: Int
    @objc public let items: [ActitoUserInboxItem]

    @objc public init(count: Int, unread: Int, items: [ActitoUserInboxItem]) {
        self.count = count
        self.unread = unread
        self.items = items
    }

    public convenience init(from response: ActitoUserInboxKit.ActitoUserInboxResponse) {
        self.init(
            count: response.count,
            unread: response.unread,
            items: response.items.map { ActitoUserInboxItem(from: $0) }
        )
    }
}
